rnd/autogpt_server/autogpt_server/blocks/basic.py

```diff
@@ -274,13 +289,15 @@ def run(self, input_data: Input) -> BlockOutput:
         Attempts to format the recorded_value using the fmt_string if provided.
         If formatting fails or no fmt_string is given, returns the original recorded_value.
         """
-        if input_data.fmt_string:
+        if input_data.format:
             try:
-                yield "output", input_data.fmt_string.format(input_data.recorded_value)
-            except Exception:
-                yield "output", input_data.recorded_value
+                fmt = re.sub(r"(?<!{){[ a-zA-Z0-9_]+}", r"{\g<0>}", input_data.format)
+                template = jinja.from_string(fmt)
+                yield "output", template.render({input_data.name: input_data.value})
```

Comment:
Do we really need jinja or `fmt.format(**{input_data.name: input_data.value})` will just do.